br/pkg/lightning/backend/kv/session.go

```diff
@@ -99,8 +99,20 @@ func (mb *kvMemBuf) Recycle(buf *bytesBuf) {
 func (mb *kvMemBuf) AllocateBuf(size int) {
 	mb.Lock()
 	size = mathutil.Max(units.MiB, int(utils.NextPowerOfTwo(int64(size)))*2)
-	if len(mb.availableBufs) > 0 && mb.availableBufs[0].cap >= size {
-		mb.buf = mb.availableBufs[0]
+	var (
+		existingBuf    *bytesBuf
+		existingBufIdx int
+	)
+	for i, buf := range mb.availableBufs {
```

Comment:
should we limit max length of `mb.availableBufs`? in case `encodeLoop` is slow down for we might loop through all availableBufs on every `AllocateBuf` in worst case

Reply:
Maybe no need to limit the max length.  There are two reasons: 
1. the allocation size is by the power of 2, starting from 1MB.  So the possible byte buffer capacities are 1MB, 2MB, 4MB, ...  Since the memory limit is typically around 256GB, and the reuse mechanism has fixed now, in the extreme case without any reuse, the available byte buffer pool's size is around 18 (1MB, 2MB, 4MB, .... 256GB), otherwise a reuse will happen .  
2. the typical allocation size will fall into only some byte buffer sizes.  Since the reuse mechanism has fixed now, the reuse will happen frequently.  So the overall `availableBufs` size is controllable.  

But it is OK to add a defending mechanism for the performance degration here.

Reply:
Fixed in a71021c4d153035f252e6464c37beaccd682f4f9